src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/OxoGReadCounts.java

```diff
@@ -81,6 +84,155 @@ public void annotate(final ReferenceContext refContext,
+        Map<Allele, MutableInt> countMap = f1r2Counts;
+        if (isF2R1(pileupElement.getRead())) {
+            countMap = f2r1Counts;
+        }
+
+        final boolean isRef = referenceAllele.basesMatch(getBasesForAlleleInRead(pileupElement, referenceAllele))
+                && !pileupElement.isBeforeDeletionStart() && !pileupElement.isBeforeInsertion();
+
+        Allele pileupAllele = null;
+        if (!isRef) {
+
+            for (Allele altAllele : altAlleles) {
+                final VariantContext.Type variantType = GATKProtectedVariantContextUtils.typeOfBiallelicVariant(referenceAllele, altAllele);
+
+                switch(variantType) {
+                    case INDEL:
+                        if (pileupElement.isBeforeInsertion()) {
+                            final int insertionLength = pileupElement.getLengthOfImmediatelyFollowingIndel();
+                            if (insertionLength == pileupElement.getLengthOfImmediatelyFollowingIndel()) {
+                                final String insertionBases = pileupElement.getBasesOfImmediatelyFollowingInsertion();
+                                // edge case: ignore a deletion immediately preceding an insertion as p.getBasesOfImmediatelyFollowingInsertion() returns null [EB]
+                                if (insertionBases != null) {
+                                    final boolean isMatch = Allele.extend(referenceAllele, insertionBases.getBytes()).basesMatch(altAllele);
+                                    if (isMatch) {
+                                        pileupAllele = altAllele;
+                                    }
+                                }
+                            }
+                        }
+
+                        if (pileupElement.isBeforeDeletionStart()){
+                            final int deletionLength = pileupElement.getLengthOfImmediatelyFollowingIndel();
+                            if ((referenceAllele.getBases().length - altAllele.getBases().length) == deletionLength) {
+                                pileupAllele = altAllele;
+                            }
+                        }
+                        break;
+                    case MNP:
+                    case SNP:
+                        if (altAllele.basesMatch(getBasesForAlleleInRead(pileupElement, altAllele))) {
+                            pileupAllele = altAllele;
+                        }
+                        break;
+                    case MIXED:
+                    case SYMBOLIC:
+                    case NO_VARIATION:
+                    default:
+                        return;
+                }
+            }
+
+        } else {
+            pileupAllele = referenceAllele;
+        }
+
+        if (pileupAllele == null) {
+            return;
+        }
+
+        if (getMeanBaseQualityForAlleleInRead(pileupElement, pileupAllele) < meanBaseQualityCutoff) {
```

Comment:
I feel like minimum base quality is a more appropriate criterion than average when deciding if alleles match.

Reply:
Done.